### PR TITLE
Go upgrade

### PR DESCRIPTION
**Pull request by Dmaddu**
### :nut_and_bolt: Description: What code changed, and why?
https://chefio.atlassian.net/browse/KINETICS-268
https://chefio.atlassian.net/browse/KINETICS-303
https://chefio.atlassian.net/browse/KINETICS-304
https://chefio.atlassian.net/browse/KINETICS-305

### :+1: Definition of Done
This pr got created on top of https://github.com/chef/automate/pull/7618 to fix pipeline errors
pipelines are green now

### :athletic_shoe: How to Build and Test the Change

Step 1:
First build automate-scaffolding-go with below command
`build components/automate-scaffolding-go`
Screenshot 2023-02-09 at 3 14 33 PM

Upload artifact, .hart file to your origin as shown below
`hab pkg upload <path of .hart file>`

Step 2: 
build all components which are dependent on Go
`local_scaffolding_origin=<your hab origin name> build components/<go dependent component name>/`
upload he .hart file to origin
Repeat build and upload process for all the Go dependant components.

Note: To figure out all components which are dependent on Go can be done by finding all the plan.sh files which includes `/automate-scaffolding-go` . 

Remember: Exclude `components/notifications-service2` since it's not being used.

Step 3:
Visit `https://bldr.habitat.sh/#/origins/<your origin>/packages`
Find all packages you build and uploaded, and copy the latest build
Screenshot 2023-02-09 at 3 31 03 PM

Step 4:
Visit `https://packages.chef.io/manifests/current/automate/latest_semver.json`
Replace all the packages which are dependant on Go, with your uploaded latest builds

Step 5:
Deploy Automate with modified `manifest.json `

Demo Video: 
https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/Ef7P3fwSW25KgdWZMKt3S1ABQ5vDXZLXiK2TAZbFmwlM-g?e=WFd3kD

OR Alternatively simply use below manifest for deployment

{
  "schema_version": "2",
  "hab_build": "core/hab/1.6.420/20211101172443",
  "build": "4.5.2",
  "version": "4.5.2",
  "hab": [
    "core/hab/1.6.521/20220603154827",
    "core/hab-sup/1.6.521/20220603161331",
    "core/hab-launcher/16260/20220603161305"
  ],
  "git_sha": "db06fb8b6eceae8d033f887e572446efa12792dd",
  "packages": [
    "aazeez/applications-service/1.0.0/20230213065556",
    "aazeez/authn-service/0.1.0/20230213065908",
    "aazeez/authz-service/0.1.0/20230213070657",
    "chef/automate-builder-api-proxy/0.1.0/20230111051525",
    "chef/automate-builder-api/0.1.0/20230111051655",
    "chef/automate-builder-memcached/1.6.12/20221031101018",
    "aazeez/automate-cds/0.1.0/20230213071115",
    "aazeez/automate-cli/0.1.0/20230213071756",
    "chef/automate-cs-bookshelf/15.4.0/20230111051659",
    "aazeez/automate-cs-nginx/15.4.0/20230213071955",
    "chef/automate-cs-oc-bifrost/15.4.0/20230111051659",
    "aazeez/automate-cs-oc-erchef/15.4.0/20230213073003",
    "chef/automate-dex/0.1.0/20230111051706",
    "chef/automate-elasticsearch/6.8.23/20220430072029",
    "chef/automate-es-gateway/0.1.0/20230111051659",
    "aazeez/automate-gateway/0.1.0/20230213074255",
    "chef/automate-ha-cluster-ctl/0.1.0/20221228105746",
    "chef/automate-ha-ctl/0.1.0/20221205072211",
    "chef/automate-ha-deployment/0.1.0/20221228110100",
    "chef/automate-ha-elasticsidecar/0.1.0/20230111052048",
    "chef/automate-ha-haproxy/2.2.14/20221031100252",
    "chef/automate-ha-opensearch/1.3.6/20230111051501",
    "aazeez/automate-ha-pgleaderchk/0.1.0/20230213074736",
    "chef/automate-ha-postgresql/13.5.0/20221031100250",
    "chef/automate-knife-ec-backup/0.1.0/20221031100258",
    "chef/automate-load-balancer/0.1.0/20221031100252",
    "chef/automate-minio/0.1.0/20230111051656",
    "chef/automate-netcat/0.1.0/20221031100250",
    "chef/automate-opensearch/1.3.6/20230111051501",
    "chef/automate-pg-gateway/0.0.1/20221031100257",
    "chef/automate-postgresql/13.5.0/20221031100252",
    "chef/automate-prometheus/0.1.0/20221031100252",
    "chef/automate-ui/2.0.0/20221031100310",
    "chef/automate-workflow-nginx/2.8.61/20221031101020",
    "chef/automate-workflow-server/2.8.61/20230111052045",
    "chef/backup-gateway/0.1.0/20230111051653",
    "aazeez/cereal-service/0.1.0/20230213075812",
    "aazeez/compliance-service/1.11.1/20230213080056",
    "aazeez/config-mgmt-service/0.1.0/20230213081332",
    "aazeez/data-feed-service/1.0.0/20230213081732",
    "chef/data-lifecycle-service/0.0.1/20191101111721",
    "aazeez/deployment-service/0.1.0/20230213073643",
    "aazeez/es-sidecar-service/1.0.0/20230213082349",
    "aazeez/event-feed-service/1.0.0/20230213082548",
    "aazeez/event-gateway/0.1.0/20230213082854",
    "aazeez/event-service/0.1.0/20230213083125",
    "aazeez/infra-proxy-service/0.1.0/20230213083352",
    "aazeez/ingest-service/0.1.0/20230213083900",
    "aazeez/license-control-service/1.0.0/20230213084148",
    "aazeez/local-user-service/0.1.0/20230213091827",
    "aazeez/nodemanager-service/1.0.0/20230213085613",
    "chef/notifications-service/1.0.0/20230111051505",
    "aazeez/pg-sidecar-service/0.0.1/20230213085807",
    "aazeez/report-manager-service/0.1.0/20230213090109",
    "aazeez/sample-data-service/0.1.0/20230213090442",
    "aazeez/secrets-service/1.0.0/20230213090651",
    "aazeez/session-service/0.1.0/20230213090915",
    "aazeez/teams-service/0.1.0/20230213091310",
    "aazeez/user-settings-service/0.1.0/20230213092053",
    "core/rsync/3.2.3/20220817235706"
  ]
}

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -14,11 +14,14 @@ linters-settings:
...
   govet:
     disable:
+      - assign
+      - atomic
       - copylocks
 
 linters:
   enable-all: true
   disable:
+    - bodyclose
```
Comment (vivek-yadav) [inline .golangci.yml:24]: why are we disabling so may linters, can we not fix them ?

File: .studiorc [hunk 2]
```diff
@@ -17,7 +17,7 @@ unset SSL_CERT_FILE
 # will need to be updated to pass certs along.
 export ES_VERSION=5.6.4
 export OPENSEARCH_URL="http://localhost:10144"
-export POSTGRESQL_URL="localhost:5432"
+export POSTGRESQL_URL="127.0.0.1:5432"
```
Comment (vivek-yadav) [inline .studiorc:20]: why is this change needed ?

File: components/automate-deployment/pkg/a1upgrade/a1commands.go [hunk 3]
```diff
@@ -39,9 +39,10 @@ type RabbitStats struct {
 // Since elasticsearch supports requesting the status of multiple snapshots in
 // one request, the top-level json object is a one-key map of "snapshots" to an
 // Array of status objects for individual snapshots:
-// {
-//   "snapshots": [ snap1_stats, snap2_stats, ... ]
-// }
+//
+//	{
+//	  "snapshots": [ snap1_stats, snap2_stats, ... ]
```
Comment (vivek-yadav) [inline components/automate-deployment/pkg/a1upgrade/a1commands.go:44]: same here, why is this comment changed ?

File: .golangci.yml (hunk 1 above)
Comment (Dmaddu) [reply .golangci.yml:24]: These all are the new linters added after 1.26.0 and on or before 1.50.1 release. Golang 19 won't support the linter of version 1.26.0. So for now we disabled some.

File: components/automate-deployment/pkg/a1stub/test_harness.go [hunk 4]
```diff
@@ -92,11 +92,12 @@ osQYj4UQRiMOwQfFY7R9rjdWHdrYeBS6BvWMu6NZY105upZfKVWhAD0=
 // StartTestHarness does what is necessary to be able to run the
 // `migrate-from-v1` code against a host that doesn't actually have A1
 // installed. The test harness consists of these things:
-// * sanity checks to make sure we actually don't have a1 installed, just in
-//   case.
-// * a set of stub binaries for commands such as `automate-ctl`;
-//   StartTestHarness puts these in the PATH
-// * a stub A1 API HTTP server.
+//   - sanity checks to make sure we actually don't have a1 installed, just in
```
[the remark this answers is not in the document]
Comment (Dmaddu) [reply components/automate-deployment/pkg/a1stub/test_harness.go:95]: bcoz of gofmt on golang 1.19

**Review by punitmundra — APPROVED** [review]
LGTM ...we can build the package and test fresh deployment, upgrade, backup and restore ...HA deployment etc...

File: lib/grpc/secureconn/factory_test.go [hunk 5]
```diff
@@ -80,7 +80,7 @@ func TestSecureConnFactory(t *testing.T) {
 
 		s, ok := status.FromError(err)
 		require.True(t, ok)
-		assert.Regexp(t, "authentication handshake failed: x509.*valid for root-a-service, not root-foo-service", s.Message())
+		assert.Regexp(t, "authentication handshake failed: x509.*valid for root-a-service, root-a-service, not root-foo-service", s.Message())
```
Comment (vivek-yadav) [inline lib/grpc/secureconn/factory_test.go:83]: why does this need to change ?

Comment (Dmaddu) [reply lib/grpc/secureconn/factory_test.go:83]: Golang 1.19 onwards, the certificate won't be valid if they do not have `Subject Alternative Name`. We have generated the new cert (lib/grpc/secureconn/testdata/root-a-service.crt) with SAN having more than one DNS. That's why we need to modify the assertion stmt.

**Review by vivek-yadav — APPROVED** [review]
